### ISSUE TITLE
Ivy Support

### ISSUE THREAD

**Issue by jeremysmartt** [Teradata/covalent#1402]

**Comment by jeremysmartt** [comment]
Design Steven
Review Julie

**Comment by iget-master** [comment]
There's any release date for v3.0.0? Angular will release v9.0.0 soon with Ivy enabled by default. This package will be useless on next angular version :-(

**Comment by jeremysmartt** [comment]
@iget-master We are planning to have a beta our soon as well as Ivy support

**Comment by emoralesb05** [comment]
Anybody can actually start testing this using the nightly build, any issues found will be appreciated!

**Comment by danielehrhardt** [comment]
> Anybody can actually start testing this using the nightly build, any issues found will be appreciated!

```
Error: Error on worker #5: Error: Tried to overwrite C:/Projekte/Webaro/webaro.cloud/node_modules/@angular/platform-browser/testing/testing.d.ts.__ivy_ngcc_bak with an ngcc back up file, which is disallowed.
    at NewEntryPointFileWriter.InPlaceFileWriter.writeFileAndBackup (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\writing\in_place_file_writer.js:36:23)
    at NewEntryPointFileWriter.writeFile (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\writing\new_entry_point_file_writer.js:63:53)
    at C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\writing\new_entry_point_file_writer.js:45:69
    at Array.forEach (<anonymous>)
    at NewEntryPointFileWriter.writeBundle (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\writing\new_entry_point_file_writer.js:45:30)
    at ClusterWorker.compile (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\main.js:166:32)
    at Worker.<anonymous> (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\worker.js:41:42)
    at Worker.emit (events.js:210:5)
    at process.<anonymous> (internal/cluster/worker.js:30:12)
    at process.emit (events.js:210:5)
    at ClusterMaster.onWorkerMessage (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:158:27)
    at C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:46:95
    at ClusterMaster.<anonymous> (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:238:57)
    at step (C:\Projekte\Webaro\webaro.cloud\node_modules\tslib\tslib.js:136:27)
    at Object.next (C:\Projekte\Webaro\webaro.cloud\node_modules\tslib\tslib.js:117:57)
    at C:\Projekte\Webaro\webaro.cloud\node_modules\tslib\tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (C:\Projekte\Webaro\webaro.cloud\node_modules\tslib\tslib.js:106:16)
    at EventEmitter.<anonymous> (C:\Projekte\Webaro\webaro.cloud\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:232:32)
    at EventEmitter.emit (events.js:210:5)
```

**Comment by Aylenor** [comment]
Ivy still doesn't work with covalent 
`WARNING in Entry point '@covalent/core' contains deep imports into 'C:/Projects/I4U/src/insereng/insereng-app/node_modules/rxjs/internal/observable/fromEvent'. This is probably not a problem, but may cause the compilation of entry points to be out of order.

ERROR in Tried to overwrite C:/Projects/I4U/src/insereng/insereng-app/node_modules/@covalent/core/paging/paging-bar.component.d.ts.__ivy_ngcc_bak with an ngcc back up file, which is disallowed.`

**Comment by Zwartpet** [comment]
Same error as @Aylenor 

```
ERROR in Tried to overwrite /Users/johnzwarthoed/projects/poc-care/node_modules/@covalent/core/paging/paging-bar.component.d.ts.__ivy_ngcc_bak with an ngcc back up file, which is disallowed.
```

With latest covalent beta and angular:
```
Angular CLI: 9.0.0-rc.12
Node: 10.16.0
OS: darwin x64

Angular: 9.0.0-rc.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.12
@angular-devkit/build-angular     0.900.0-rc.12
@angular-devkit/build-optimizer   0.900.0-rc.12
@angular-devkit/build-webpack     0.900.0-rc.12
@angular-devkit/core              9.0.0-rc.12
@angular-devkit/schematics        9.0.0-rc.12
@angular/cdk                      9.0.0-rc.9
@angular/cli                      9.0.0-rc.12
@angular/flex-layout              9.0.0-beta.28
@angular/material                 9.0.0-rc.9
@ngtools/webpack                  9.0.0-rc.12
@schematics/angular               9.0.0-rc.12
@schematics/update                0.900.0-rc.12
rxjs                              6.5.3
typescript                        3.7.5
webpack                           4.41.2
```

**Comment by priyankpat** [comment]
Also having similar issue

```
ERROR in Tried to overwrite /Users/shankz/Documents/RSPNDR/rspndr-toolbox/node_modules/@covalent/core/common/forms/auto-trim/auto-trim.directive.d.ts.__ivy_ngcc_bak with an ngcc back up file, which is disallowed.
```

Version info:
```
Angular CLI: 9.0.0-rc.12
Node: 12.12.0
OS: darwin x64

Angular: 9.0.0-rc.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router, service-worker
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.23
@angular-devkit/build-angular     0.900.0-rc.13
@angular-devkit/build-optimizer   0.900.0-rc.13
@angular-devkit/build-webpack     0.900.0-rc.13
@angular-devkit/core              7.3.9
@angular-devkit/schematics        7.3.9
@angular/cdk                      9.0.0-rc.9
@angular/cli                      9.0.0-rc.12
@angular/flex-layout              9.0.0-beta.28
@angular/material                 9.0.0-rc.9
@ngtools/webpack                  9.0.0-rc.13
@schematics/angular               8.3.23
@schematics/update                0.900.0-rc.12
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2
```

**Comment by emoralesb05** [comment]
@christianmemije Have you seen this error?

**Comment by christianmemije** [comment]
I ran into the `ERROR in Tried to overwrite` issue a few weeks ago. Just now, I attempted to reproduce using the following versions, and the project was able to compile successfully. @Zwartpet  & @priyankpat, can yall try to use these versions and see if the error persists?
```
Angular CLI: 9.0.0-rc.13
Node: 12.13.1
OS: darwin x64

Angular: 9.0.0-rc.13
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.13
@angular-devkit/build-angular     0.900.0-rc.13
@angular-devkit/build-optimizer   0.900.0-rc.13
@angular-devkit/build-webpack     0.900.0-rc.13
@angular-devkit/core              9.0.0-rc.13
@angular-devkit/schematics        9.0.0-rc.13
@angular/cdk                      9.0.0-rc.9
@angular/material                 9.0.0-rc.9
@ngtools/webpack                  9.0.0-rc.13
@schematics/angular               9.0.0-rc.13
@schematics/update                0.900.0-rc.13
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2
```

**Comment by Zwartpet** [comment]
Issue still persists with:
```
Angular CLI: 9.0.0-rc.14
Node: 10.16.0
OS: darwin x64

Angular: 9.0.0-rc.14
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.14
@angular-devkit/build-angular     0.900.0-rc.14
@angular-devkit/build-optimizer   0.900.0-rc.14
@angular-devkit/build-webpack     0.900.0-rc.14
@angular-devkit/core              9.0.0-rc.14
@angular-devkit/schematics        9.0.0-rc.14
@angular/cdk                      9.0.0-rc.9
@angular/flex-layout              9.0.0-beta.28
@angular/material                 9.0.0-rc.9
@ngtools/webpack                  9.0.0-rc.14
@schematics/angular               9.0.0-rc.14
@schematics/update                0.900.0-rc.14
rxjs                              6.5.3
typescript                        3.7.5
webpack                           4.41.2
```

**Comment by Zwartpet** [comment]
Could be a difference between covalent 3.0.0-beta.2 i'm using and the develop branch? I see some things have changed but don't know if it will fix the issue.

**Comment by Zwartpet** [comment]
Can't seem to reproduce the issue with the covalent project, but if I install covalent in any other project with npm it won't build

**Comment by Zwartpet** [comment]
I believe https://github.com/Teradata/covalent/commit/f0dd4c652d61db957f6296c6a9a18871fab1782f fixed the issue, any idea when/if a new beta will be created?

**Comment by emoralesb05** [comment]
We will do some preliminary testing with our internal products and do a beta.2 if all goes well tomorrow. Hopefully it goes smoothly.

**Comment by emoralesb05** [comment]
To everyone in the thread, Please help us by testing with the nightly build and tell us if any other issues come up

**Comment by christianmemije** [comment]
For reference, to use nightly:
```
"@covalent/core": "git+https://github.com/Teradata/covalent-nightly.git",
```

**Comment by Zwartpet** [comment]
Building with the nightly works 👍 
Thanks for the quick/great support

**Comment by Zwartpet** [comment]
Angular 9 was released, might as well make it stable 3.0 release ;)

**Comment by emoralesb05** [comment]
Yup, we are gonna start our RC process this week, please try our beta.3 and let us know if there are any issues 😄 

Closing this since we should support ivy and angular 9 with our latest `3.0.0-beta.3` release
